app/models/user.rb

```diff
@@ -1,13 +1,73 @@
 class User < ActiveRecord::Base
+  
+  attr_accessor :remember_token, :activation_token, :reset_token 
+
+  before_save :downcase_email
+  before_create :create_activation_digest
+
   has_many :posts, dependent: :destroy
   has_secure_password
 
-  before_save { self.email = email.downcase }
-
   validates :name, presence: true, length: { maximum: 50 }
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
   validates :email, presence: true, length: { maximum: 255 },
-    format: { with: VALID_EMAIL_REGEX },
-    uniqueness: { case_sensitive: false }
+            format: { with: VALID_EMAIL_REGEX },
+            uniqueness: { case_sensitive: false }
   validates :password, presence: true, length: { minimum: 6 }
-end
+
+  def User.new_token
+    SecureRandom.urlsafe_base64
+  end 
+
+  def User.digest(string)
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+                                                  BCrypt::Engine.cost
+    BCrypt::Password.create(string, cost: cost)
+  end 
+
+  def authenticated?(attribute, token)
+    digest = send("#{attribute}_digest")
+    return false if digest.nil? 
+    BCrypt::Password.new(digest).is_password?(token)
+  end 
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
+  end 
+
+  def forget 
+    update_attribute(:remember_digest, nil)
+  end 
```

Comment:
Cool, I've looked into PORO's and what they're valuable for. Should I attempt to go even further and break those additional methods into even more individual classes? (e.g. UserActivation, UserAuthentication, UserCookies, etc. Or is it good enough to simply store all of those methods into a catch-all class like you've mentioned?)